Chat/TokenManagement/UserTokenManager.cs

```diff
@@ -28,5 +29,26 @@ public async Task<(string userMri, string skypeToken, long expiresDateUtc)> Gene
                 return (null, null, -1);
             }
         }
+
+        public async Task<(string userMri, string skypeToken, long expiresDateUtc)> RefreshTokenAsync(string resourceConnectionString, string identity)
+        {
+            try
+            {
+                CommunicationIdentityClient communicationIdentityClient = new CommunicationIdentityClient(resourceConnectionString);
+                CommunicationUser user = new CommunicationUser(identity);
+
+                Azure.Response<Azure.Communication.Administration.Models.CommunicationUserToken> tokenResponse =
+                    await communicationIdentityClient.IssueTokenAsync(user, scopes: new[] { CommunicationTokenScope.Chat });
+
+                string token = tokenResponse.Value.Token;
+                DateTimeOffset expiresOn = tokenResponse.Value.ExpiresOn;
+                return (identity, token, expiresOn.Ticks);
+            }
+            catch (Azure.RequestFailedException ex)
+            {
+                Console.WriteLine($"Error occured while Generating Token: {ex}");
+                return (null, null, -1);
```

Comment:
Wouldn't it be better to throw than return invalid values?

Reply:
converting to throws